src/main/java/uk/gov/hmcts/reform/wataskconfigurationapi/services/configurators/CamundaProcessVariableExtractor.java

```diff
@@ -0,0 +1,26 @@
+package uk.gov.hmcts.reform.wataskconfigurationapi.services.configurators;
+
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.wataskconfigurationapi.domain.entities.camunda.CamundaValue;
+import uk.gov.hmcts.reform.wataskconfigurationapi.domain.entities.camunda.TaskResponse;
+import uk.gov.hmcts.reform.wataskconfigurationapi.services.ConfigureTaskService;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@SuppressWarnings({"PMD.UseConcurrentHashMap"})
+@Component
+@Order(2)
+public class CamundaProcessVariableExtractor implements TaskConfigurator {
+
+    @Override
+    public Map<String, Object> getConfigurationVariables(TaskResponse task,
+                                                         Map<String, CamundaValue<Object>> processVariables) {
+        CamundaValue<Object> caseId = processVariables.get(ConfigureTaskService.CASE_ID_PROCESS_VARIABLE_KEY);
+        Map<String, Object> mappedDetails = new HashMap<>();
```

Comment:
Not entirely sure, but cannot we fix that PMD issue if you use the concurrent hash map?

Reply:
Yes you're right I'll include this in the next PRs since I want to keep this one as only a package refactoring